sdk/eventhub/event-hubs/src/dataTransformer.ts

```diff
@@ -17,21 +29,25 @@ export const defaultDataTransformer = {
    * and returns an encoded body (some form of AMQP type).
    *
    * @param body - The AMQP message body
-   * @returns The encoded AMQP message body as an AMQP Data type
-   * (data section in rhea terms). Section object with following properties:
-   * - typecode: 117 (0x75)
-   * - content: The given AMQP message body as a Buffer.
-   * - multiple: true | undefined.
+   * @param bodyType - The AMQP section to story the body in.
+   * @returns The encoded AMQP message body as an AMQP Data/Sequence/Value section.
    */
-  encode(body: unknown): any {
+  encode(body: unknown, bodyType: BodyTypes): any {
     let result: any;
-    if (isBuffer(body)) {
+    // string, undefined, null, boolean, array, object, number should end up here
+    // coercing undefined to null as that will ensure that null value will be given to the
+    // customer on receive.
+    if (body === undefined) body = null;
```

Comment:
I'd throw in a blank line here - it's really easy to think the 'if' and the 'underlying else's' all are part of the same chain.

With that said - the 'body being null' thing seems incorrect to me. I know this came from SB, so it's potentially wrong there as well.

For instance, if you pass `null` you get this from JSON.stringify:

```bash
> Buffer.from(JSON.stringify(null), "utf8")
<Buffer 6e 75 6c 6c>

> Buffer.from(JSON.stringify(null), "utf8").toString()
'null'
```
I haven't looked, but perhaps this all plays nicely with the rhea encoding method for `message.<data|value|sequence>_section()` methods?

Reply:
The only time `null` would have some transformation done on it is for the `data_section`, which case it will be JSON stringified. I'd agree that this would cause a problem, except we also attempt to JSON parse any data_section bodies. This would result in the body being `null`, not `"null"`.

That said, we might run into a problem is when talking to the other languages. There's no guarantee other languages are also calling JSON.parse (in fact I doubt they are), so they may see 'null' as a string rather than a null type. It looks like this is already existing behavior, but since `null` and `'null'` both result in `null` being returned from JSON.parse, updating this is safe.